File: Cpp/1232_Check_If_It_Is_a_Straight Line.cpp
class Solution {
public:
    bool checkStraightLine(vector<vector<int>>& coordinates) {
        if(coordinates.size()<2) return true;

        for(int i=2; i<coordinates.size(); ++i){
            if(getDiffX(coordinates[i], coordinates[i-1]) * getDiffY(coordinates[i-1], coordinates[i-2])
               != getDiffY(coordinates[i], coordinates[i-1]) * getDiffX(coordinates[i-1], coordinates[i-2])) return false;
        }
        
        return true;
    }
    
    float getDiffX(vector<int> p1, vector<int> p2){
        return p1[0] - p2[0];
    }
    
    float getDiffY(vector<int> p1, vector<int> p2){
        return p1[1] - p2[1];
    }
};

File: Kotlin/1232_Check_If_It_Is_a_Straight_Line.kt
class Solution {
    fun checkStraightLine(coordinates: Array<IntArray>): Boolean {
        for(i in 2 until coordinates.size) {
            if(getDiffX(coordinates[i], coordinates[i-1]) * getDiffY(coordinates[i-1], coordinates[i-2])
               != getDiffY(coordinates[i], coordinates[i-1]) * getDiffX(coordinates[i-1], coordinates[i-2])) return false
        }
        
        return true
    }
    
    fun getDiffX(p1: IntArray, p2: IntArray): Int {
        return p1[0] - p2[0]
    }
    
    fun getDiffY(p1: IntArray, p2: IntArray): Int {
        return p1[1] - p2[1]
    }
}